### PR TITLE
Solr datasource

### PR DESCRIPTION
**Pull request by ktyborowski**
This is an implementation for Apache Solr as a data source. It's based on a JDBC connection and python JayDeBeApi.

Thought it might be nice to have a direct connection to Solr inside MindsDB. Solr in some cases has it's own SQL syntax so this should be considered. This implementation also assumes that the solr jdbc connector is in the CLASSPATH. Java also needs to be installed in the environment.

No tests added for now as the idea of a Solr datasource needs to be discussed with the maintainers. The tests could use a base Solr installation though, as it comes with a default collection to query against.

Changes:

1. Added a Solr datasource
2. Added JayDeBeApi to requirements.txt

If this gets approved it would be nice if it counted towards the hacktoberfest goal 😄

### CODE REVIEW

File: mindsdb_datasources/datasources/solr_ds.py [hunk 1]
```diff
@@ -0,0 +1,24 @@
+import pandas as pd
+import jaydebeapi
+from mindsdb_datasources.datasources.data_source import SQLDataSource
+
+class SolrDS(SQLDataSource):
```
Comment (ZoranPandovski) [inline mindsdb_datasources/datasources/solr_ds.py:5]: Please just add this to __init__ py

**Review by ZoranPandovski — CHANGES_REQUESTED** [review]

File: requirements.txt [hunk 2]
```diff
@@ -11,3 +11,4 @@ moz_sql_parser >= 3.0.0, < 4.1.20329
 pg8000 >= 1.15.3
 pymongo[tls, srv] >= 3.10.1
 certifi
+JayDeBeApi 1.2.3
```
Comment (ZoranPandovski) [inline requirements.txt]: Will this work without == ?

**Review by ZoranPandovski — CHANGES_REQUESTED** [review]

File: requirements.txt (hunk 2 above)
Comment (ktyborowski) [reply requirements.txt]: Hey, I looked into snowflake datasource implementation and changed accordingly. For now I have removed the requirements and added information in the init. The init info includes the jaydebeapi , the solr install and the classpath export requirements. 

One more thing with the classpath exporting is that it is possible to give the path to the JDBC driver jar. Would you like to add an optional parameter for that? If this parameter is not set the Solr datasource could fallback to the CLASSPATH variable.

File: mindsdb_datasources/datasources/solr_ds.py (hunk 1 above)
Comment (ktyborowski) [reply mindsdb_datasources/datasources/solr_ds.py:5]: Added.

**Comment by ZoranPandovski** [comment]
****CLA Assistant Lite bot**** All contributors have signed the CLA  ✍️

**Comment by ktyborowski** [comment]
Responding to the CLA agreement.

I have read the CLA Document and I hereby sign the CLA.

**Comment by ZoranPandovski** [comment]
@pixpack Please just respond with `I have read the CLA Document and I hereby sign the CLA.` otherwise the bot will not be triggered :)

**Comment by ktyborowski** [comment]
I have read the CLA Document and I hereby sign the CLA.

**Pull request merged**